### ISSUE TITLE
netty-socketio 2.0.3 是不是不支持socket.io-client 3.0+

### ISSUE THREAD

**Issue by qq1031824970** [mrniko/netty-socketio#935]
我看到好像支持了socket.io-client4。但是使用时发现用socket.io-client3.0+版本发送二进制数据时后端会出现解码错误，错误如下：
![image](https://github.com/mrniko/netty-socketio/assets/85598470/a894052f-02af-477e-9268-265ae3964bb5)
![image](https://github.com/mrniko/netty-socketio/assets/85598470/7635c2e3-4f59-4598-9d13-69e8f5963b6c)
二进制传输是还没有适配吗

**Comment by qq1031824970** [comment]
使用2.5.0版本可以正常传输，这好像也是官方demo中使用的js版本

**Comment by qq1031824970** [comment]
![image](https://github.com/mrniko/netty-socketio/assets/85598470/5129445e-cc94-447b-82ae-5363d8d618f2)
netty-socketio 2.0.3   socket.io server version maybe 2.x

**Comment by mrniko** [comment]
netty-socketio demo is out of date

**Comment by MindaDai** [comment]
兄弟解决了吗，2+版本前端js版本用多少啊

**Comment by kyan54** [comment]
> 兄弟解决了吗，2+版本前端js版本用多少啊

建议别用2+的版本，前端流浪器放到非活动页就经常掉线
我现在用的就是后端是这边的最新版本，前端用的2.2版本，听说4版本解决了这个问题不知道真假

**Comment by xuhaoyang** [comment]
The current test results indicate that V3/V4 cannot be supported.

The Swift version, need to force setting V2 option.
When the Java client version connect, it triggers the server's `addConnectListener` twice.
The Dart client version 2.x cannot connect and needs to downgrade to 1.x.
For the other unmentioned languages, when sending binary data, an extra byte (value is 4) will be added at the beginning.
---------------
目前测试下来是无法正常支持V3和V4的client端。

Swift Client端需要强制设置V2
Java Client端连接时会触发两次服务端的`addConnectListener`
Dart Client端2.x版本无法连接，需要降级1.x
在可以连接的client端，二进制会多首位多增加一位byte(值是4)

**Comment by bulinbulinxmz** [comment]
I have discovered two issues when using nettysocketio 2.0.8 with EIOv4 (socket.io-client@4.7.3) which does not support binary.
1.When sending binary data using V4, it does not prepend a byte with the value 4 (indicating binary data). However, nettysocketio processes by checking the first bit for this identifier. 
Therefore, before sending, the identifier needs to be added.
2. When receiving binary data using V4, nettysocketio adds a byte with the identifier 4 at the very beginning of the binary data.
00000000: **04**0a 1531 4852 7841 314f 5950 566e 5137  ...1HRxA1OYPVnQ7
00000001: 3376 674c 4970 7652 108b c8f6 8dd7 311a  3vgLIpvR......1.
00000002: 170a 0a31 3030 3031 3130 3030 3112 09e5  ...1000110001...
00000003: a4a7 e790 86e7 9fb3                      ........

In conclusion, when using nettysocketio 2.0.8 in combination with EIOv4 to send and receive binary data, it is necessary to manually handle the identification bit of the binary data.
